test/unit/core/metafunction/basic_test.cpp

```diff
@@ -0,0 +1,65 @@
+//
+// Copyright (c) 2006-2018, Knut Reinert & Freie Universitaet Berlin
+// Copyright (c) 2016-2018, Knut Reinert & MPI Molekulare Genetik
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+#include <list>
+#include <vector>
+
+#include <gtest/gtest.h>
+
+#include <meta/meta.hpp>
+
+#include <range/v3/view/iota.hpp>
+#include <range/v3/view/take_exactly.hpp>
+
+#include <seqan3/core/metafunction/basic.hpp>
+#include <seqan3/range/detail/random_access_iterator.hpp>
+
+
+using namespace seqan3;
+
+TEST(metafunction, remove_cvref_)
+{
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int const>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int volatile>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int &>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int &&>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int const &>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int const &&>>));
+    EXPECT_TRUE((std::is_same_v<int, remove_cvref_t<int const volatile &&>>));
+    // don't decay pointers and arrays:
+    EXPECT_FALSE((std::is_same_v<int, remove_cvref_t<int*>>));
+    EXPECT_FALSE((std::is_same_v<int, remove_cvref_t<int[3]>>));
+    EXPECT_FALSE((std::is_same_v<int*, remove_cvref_t<int[3]>>));
```

Comment:
What are the types of those? Can you give positive test cases instead of negative ones?